unit_test/solver/network/utils/test_fluid_properties.py

```diff
@@ -15,196 +15,183 @@
-        self.assertEqual(mock_csv_reader.call_count, 1)
-
+        self.assertEqual(interpolation.x, x)
+        self.assertEqual(interpolation.y, y)
+        self.assertEqual(interpolation.order, order)
+        npt.assert_almost_equal(interpolation.coefficients, np.array([2, 0]))
 
-class FluidPropertiesTest(unittest.TestCase):
-    """Testcase for FluidProperties class."""
-
-    @patch("csv.reader")
-    def setUp(self, mock_csv_reader):
-        """Set up the test."""
-        mock_csv_reader.return_value = [["1", "2", "3", "4", "5"], ["6", "7", "8", "9", "10"]]
-
-        self.fluid_props = FluidProperties()
-
-    def test_get_internal_energy(self):
-        """Test the get_internal_energy method."""
+    def test_init_order_5(self):
+        """Test the initialization of the Interpolation class."""
         # Arrange
-        T = 1
+        x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        order = 5
+        y = [value**order for value in x]
 
         # Act
-        IE = self.fluid_props.get_ie(T)
+        interpolation = Interpolation(x, y, order)
 
         # Assert
-        self.assertEqual(IE, 0)
+        self.assertEqual(interpolation.x, x)
+        self.assertEqual(interpolation.y, y)
+        self.assertEqual(interpolation.order, order)
+        npt.assert_almost_equal(interpolation.coefficients, np.array([1, 0, 0, 0, 0, 0]))
 
-    def test_get_internal_energy_with_invalid_temperature(self):
-        """Test the get_internal_energy method with an invalid temperature."""
+    def test_call(self):
+        """Test the call method of the Interpolation class."""
         # Arrange
-        T = 0
+        x = [1, 2, 3]
+        y = [2, 4, 6]
+        order = 1
+        interpolation = Interpolation(x, y, order)
 
         # Act
-        with self.assertRaises(ValueError) as cm:
-            self.fluid_props.get_ie(T)
+        interpolated_value = interpolation(2)
 
         # Assert
-        self.assertEqual(
-            str(cm.exception), "The temperature 0 is outside the range of the fluid properties."
-        )
-        self.assertIsInstance(cm.exception, ValueError)
+        self.assertAlmostEquals(interpolated_value, 4)
 
-    def test_get_temperature(self):
-        """Test the get_temperature method."""
+    def test_call_order_5(self):
+        """Test the call method of the Interpolation class."""
         # Arrange
-        IE = 0
+        x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        order = 5
+        y = [value**order for value in x]
+        interpolation = Interpolation(x, y, order)
 
         # Act
-        T = self.fluid_props.get_t(IE)
+        interpolated_value = interpolation(2)
 
         # Assert
-        self.assertEqual(T, 1)
+        self.assertAlmostEquals(interpolated_value, 32)
 
-    def test_get_temperature_with_invalid_internal_energy(self):
-        """Test the get_temperature method with an invalid internal energy."""
+    def test_validation_error(self):
```

Comment:
Maybe include a test with a polynomial y = ax^2+bx+c with just some data x = [0, 1E3] and some random noise?

Reply:
I include test with complete polynomial, but no noise added.